hooks/checkout

```diff
@@ -13,39 +25,62 @@ check_set() {
   fi
 }
 
+# Checks for failed exit codes returned by a timeout command.
+# If the underlying command fails due to timeout, this function stops the current execution returning either the
+# TIMEOUT exit code (124) or the override value provided by the caller.
+# In any other non-timeout failure scenarios, the current execution is stopped but the underlying command's exit
+# code is returned unmodified.
+# Arguments:
+#   $1: The exit code returned by the timeout command.
+#   $2: The exit code to be returned when the underlying command times out.
+check_timeout_exit_code() {
+  local exit_code="$1"
+  local timeout_exit_code_override="${2:-}"
+  if [[ "${exit_code}" -eq 124 ]] && [[ -n "${timeout_exit_code_override}" ]]; then
```

Comment:
I think you can combine this.
```suggestion
  if [[ "${exit_code}" -eq 124 && -n "${timeout_exit_code_override}" ]]; then
```

Reply:
I tend to find the current approach a bit more readable but since you are the repo owner I will update as you suggest 😄